File: day1/part2.jl
#!/usr/bin/env julia

using DelimitedFiles

depths = readdlm("input.txt", '\n', Int)

inc_cnt = 0;
for i in 4:length(depths)
    if depths[i] + depths[i-1] + depths[i-2] > 
       depths[i-1] + depths[i-2] + depths[i-3]
        global inc_cnt += 1
    end
end

print(inc_cnt)

# actual function!
function f(depths, sample_size)
    inc_count = 0;
    for i in 1:(length(depths)-sample_size)
        if sum(depths[i+1:i+sample_size]) > sum(depths[i:i+sample_size-1])
            inc_count += 1
        end
    end
    return inc_count
end

print(f(depths, 3))

File: day1/part2.py
#!/usr/bin/env python3

with open('input.txt') as f:
    depths = f.read().splitlines()

depth_list = []
for depth in depths:
    depth_list.append(int(depth))

inc_count = 0
for i, depth in enumerate(depth_list):
    if depth_list[i] + depth_list[i-1] + depth_list[i-2] > \
       depth_list[i-1] + depth_list[i-2] + depth_list[i-3]:
        inc_count += 1

print(inc_count)    

# actual function for this
def depth_increaser(depths, sample_size):
    inc_count = 0
    for i in range(len(depths)-sample_size):
        if sum(depths[i:i+sample_size]) > sum(depths[i-1:i-1+sample_size]):
            inc_count +=1
    return inc_count

print(depth_increaser(depth_list, 3))